$NetBSD$

Use bundled boost, but system OpenSSL and zlib.
Fix mysqlxtest binary linking.

--- rapid/plugin/x/CMakeLists.txt.orig	2018-12-18 15:24:44.000000000 +0000
+++ rapid/plugin/x/CMakeLists.txt
@@ -92,13 +92,13 @@ ELSE()
     ${CMAKE_SOURCE_DIR}/include #temporary
     ${CMAKE_CURRENT_SOURCE_DIR}/include/mysql
     ${CMAKE_CURRENT_BINARY_DIR}/generated
-    ${SSL_INCLUDE_DIRS}
+    ${BOOST_INCLUDE_DIR}
   )
 
   INCLUDE_DIRECTORIES(SYSTEM
-    ${BOOST_INCLUDE_DIR}
     ${PROTOBUF_INCLUDE_DIRS}
     ${LIBEVENT_INCLUDE_DIR}
+    ${SSL_INCLUDE_DIRS}
   )
 
   IF(XPLUGIN_NO_LITE_PROTOBUF)
@@ -214,7 +214,10 @@ ELSE()
   ENDIF()
 
   TARGET_LINK_LIBRARIES(mysqlxtest mysys strings perconaserverclient)
-  TARGET_LINK_LIBRARIES(mysqlxtest ${PROTOBUF_LIBRARY})
+  TARGET_LINK_LIBRARIES(mysqlxtest ${PROTOBUF_LIBRARY} ${CRYPTO_LIBRARY})
+  IF(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
+    TARGET_LINK_LIBRARIES(mysqlxtest "socket")
+  ENDIF()
 
   IF(INSTALL_MYSQLTESTDIR)
     INSTALL(
